cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -1210,6 +1210,49 @@ aggregate_reader_metadata::apply_row_bounds_filter(
                     std::move(row_group_row_offsets)};
 }
 
+std::vector<std::vector<size_type>> aggregate_reader_metadata::apply_byte_bounds_filter(
+  host_span<std::vector<size_type> const> input_row_group_indices,
+  size_t bytes_to_skip,
+  std::optional<size_t> const& bytes_to_read) const
+{
+  CUDF_EXPECTS(input_row_group_indices.size() == 1,
+               "Byte bounds filter can only be applied to a single source",
+               std::invalid_argument);
+
+  auto filtered_row_group_indices =
+    std::vector<std::vector<size_type>>(input_row_group_indices.size());
+
+  std::for_each(
+    input_row_group_indices.front().begin(),
+    input_row_group_indices.front().end(),
+    [&](auto const& rg_idx) {
+      // Get the file offset of this row group
+      auto const row_group_file_offset = [&]() {
+        auto const& rg = per_file_metadata.front().row_groups[rg_idx];
+        if (rg.file_offset.has_value()) {
+          return rg.file_offset.value();
+        } else if (rg.columns.front().file_offset != 0) {
+          return rg.columns.front().file_offset;
+        } else {
+          auto const& col_meta = rg.columns.front().meta_data;
+          return col_meta.dictionary_page_offset != 0
+                   ? std::min(col_meta.dictionary_page_offset, col_meta.data_page_offset)
+                   : col_meta.data_page_offset;
+        }
+      }();
+
+      // Check if it is within the specified byte range
+      if (std::cmp_greater_equal(row_group_file_offset, bytes_to_skip) and
+          (bytes_to_read.has_value()
+             ? std::cmp_less(row_group_file_offset, bytes_to_skip + bytes_to_read.value())
```

Comment:
Should we check `bytes_to_read.value() + bytes_to_skip` would overflow `size_t` since it's possible someone may pass a `std::numeric_limits<size_t>::max()` thinking they don't want any byte limit.